hot-reload.md

```diff
@@ -0,0 +1,281 @@
+  }
+}
+```
+
+After running this app, you might change the code as follows:
+
+```
+import 'package:flutter/widgets.dart';
+void main() {
+  runApp(
+    const Center(
+      child: const Text('Hello', textDirection: TextDirection.ltr)));
+  }
+```
+
+
+With a full restart, the program starts from the beginning, executes the new
+version of `main()`, and builds a widget tree that displays the text `Hello`.
+
+However, if you hot reload the app after this change, `main()` is not 
+re-executed, and the widget tree is rebuilt with the unchanged instance of 
+`MyApp` as the root widget. The result is no visible change after hot reload.
+
+
+### Recent code change is included but app state is excluded
+
+In Dart, [static fields are lazily initialized](https://news.dartlang.org/2012/02/static-variables-no-longer-have-to-be.html). This means that the first time you run a Flutter app and a static field is read,
+it is set to whatever value its initializer was evaluated to. Thus if you change
+initializers of global variables and static fields, those changes are not
+visible after hot reload. A full restart is necessary for this type of change
+to take effect. 
+
+For example, consider the following code:
+
+```
+final sampleTable = [
+  new Table("T1"),
+  new Table("T2"),
+  new Table("T3"),
+  new Table("T4"),
+];
+```
+After running the app, if you make the following change:
+```
+final sampleTable = [
+  new Table("T1"),
+  new Table("T2"),
+  new Table("T3"),
+  new Table("T10"),    //modified
+];
+```
+and then hot reload, the change is not reflected. 
+
+Conversely, in the following example:
+```
+const foo = 1;
+final bar = foo;
+print(foo);
+print(bar);
+```
+
+
+running the app for the first time prints `1` and `1`. Then if you make the
+following change: 
+
+```
+const foo = 2;    //modified
+final bar = foo;
+print(foo);
+print(bar);
+```
+and hot reload, it now prints `2` and `1`. While changes to `const` field values
+are always hot reloaded, the static field initializer is not rerun.
+Conceptually, `const` fields are treated like aliases instead of state.
+
+The Dart VM detects initializer changes and flags when a set of changes needs a
+full restart to take effect. The flagging mechanism is triggered for most of the
+initialization work in the above example, but not for cases like:
+
+```
+final bar = foo;
+```
+
+To be able to update `foo` and view the change after hot reload, consider
+redefining the field as `const` or using a getter to return the value, rather
+than using `final`. For example:
+
+```
+const bar = foo;
+```
+or:
+
+```
+get bar => foo;
+```
+
+Read more about the [differences between the `const` and `final` keywords](https://news.dartlang.org/2012/06/const-static-final-oh-my.html) in Dart. 
+
+
+### Previous state is combined with new code 
+
+If code changes affect the state of your app (or its dependencies), the Flutter
+framework gives you two options for reconciling old data with new code: throw
```

Comment:
Maybe drop "framework"? The implementation is split across the command line tools, the engine and the framework.

Reply:
Good point, done.